awscli/clidriver.py

```diff
@@ -166,7 +172,7 @@ def main(self, args):
     def _do_main(self, args):
         driver = self._driver
         if driver is None:
-            driver = create_clidriver()
+            driver = create_clidriver(args)
```

Comment:
So one thing, I think this change breaks is any command that can accept a non-existent profile, no longer works. For example, I now get this error:
```
$ aws configure set foo bar --profile not-here

The config profile (not-here) could not be found
```
To get this working with profile, I think we will have to rethink how we do auto-prompt resolution.